Project1/CellEdge.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+class GridCell;
+#include <array>
+#include "MobilityList.h"
+#include <optional>
+/**
+Represents an Edge to a GridCell, basically a struct
+*/
+class CellEdge {
+private:
+	const MobilityList<std::optional<int>> _costs = MobilityList<std::optional<int>>();
+	bool canPass(bool intangible) const;
+public:
+	CellEdge(const CellEdge& cell) = default;
+	CellEdge(CellEdge&& cell) = default;
+
+	GridCell& _cell;
```

Comment:
I haven't made this const because this is the only place a GridCells can see Their neighbors.  If we want to be able to modify a Cell's neighbor, then this must stay non-const, or GridCell needs another way to access it's neighbors.